File: jput-core/src/main/java/cz/encircled/jput/trend/TrendResult.java
package cz.encircled.jput.trend;

/**
 * @author Vlad on 27-May-17.
 */
public class TrendResult {

    public boolean isAverageMet = true;

    public long runAverageTime;

    public long standardAverage;

    public long deviation;

    public int[] notMetPercentiles;

    public boolean isError() {
        return !isAverageMet || (notMetPercentiles != null && notMetPercentiles.length > 0);
    }
}

File: jput-core/src/main/java/cz/encircled/jput/trend/TrendResult.kt
package cz.encircled.jput.trend

/**
 * @author Vlad on 27-May-17.
 */
data class TrendResult(
        var isAverageMet: Boolean = true,
        var executionAvgTime: Long = 0,
        var sampleAvg: Long = 0,
        var deviation: Long = 0) {

    val isError: Boolean
        get() = !isAverageMet

}
